### PR TITLE
Fixes #75: Adapter for Firebase Remote Config 

### LINKED ISSUE #75: Adapter for `firebaseremoteconfig.googleapis.com`
Endpoint URL: `https://firebaseremoteconfig.googleapis.com/v1/projects/*/namespaces/firebase:fetch`

Sample requests:

* [informed-consent,249697](https://data.tweasel.org/data/requests/informed-consent,249697)
* [monkey-july-2023,118033](https://data.tweasel.org/data/requests/monkey-july-2023,118033)
* [monkey-april-2024,2629](https://data.tweasel.org/data/requests/monkey-april-2024,2629)
* [worrying-confessions,40016](https://data.tweasel.org/data/requests/worrying-confessions,40016)
* [do-they-track,58615](https://data.tweasel.org/data/requests/do-they-track,58615)

### PR DESCRIPTION
**Pull request by baltpeter**
This is based on #59, which needs to be merged first.

### CODE REVIEW

File: src/adapters/google.ts [hunk 1]
```diff
@@ -1315,4 +1315,153 @@ export const adapters: Adapter[] = [
...
+                },
+                {
+                    context: 'body',
+                    path: 'analyticsUserProperties.uuid',
+                    reasoning: 'obvious property name',
+                },
+                {
+                    context: 'body',
+                    path: 'appInstanceIdToken',
+                    reasoning: 'google/appInstanceIdToken.md',
```
Comment (zner0L) [inline src/adapters/google.ts:1432]: Same question here. I think we should decode the JWT.

**Review by zner0L — CHANGES_REQUESTED** [review]

**Comment by zner0L** [comment]
Created #79 to track JWT decoding and will hold merging until this is closed.

**Comment by baltpeter** [comment]
This is now based on https://github.com/tweaselORG/TrackHAR/pull/87, which needs to be merged first.

**Comment by baltpeter** [comment]
The problem this still has is that there is [one app](https://data.tweasel.org/data?sql=select+link%28dataset%2C+id%29%2C+initiator%2C+platform%2C+runType%2C+startTime%2C+method%2C+httpVersion%2C+endpointUrl%2C+scheme%2C+host%2C+port%2C+path%2C+content%2C+headers%2C+cookies+%0D%0Afrom+requests%0D%0AWHERE+path+like+%27%25%2Fv1%2Fprojects%2F330194062051%2Fnamespaces%2Ffirebase%3Afetch%25%27%0D%0Alimit+101+collate+nocase%3B) that [sometimes](https://data.tweasel.org/data/requests/do-they-track,98586) (but not always) sends an `appInstanceIdToken` that is not a JWT: `eCKZbsz5Sde-1eY2P48xBZ:APA91bExD9TtElNISzzjwPLvFRAypKO89dpkU4SGHI4dZpu0thydzo7QqrVnE3gXg6qrb8nCay2F45__K2oL-PHTqy-VeN20yUdbDE7Eym5oFyyPqh_mykAyrnV4Y8eXl713Bg2GMK3r`

This touches on a question that I have been toeing around for a while. Currently, we don't guard against exceptions in decode functions:

https://github.com/tweaselORG/TrackHAR/blob/9287bf182bf18badb8ac93f0e300eec0b4f011c1/src/index.ts#L258

But I am really not sure whether that is "correct". For one, this a problem from the perspective of writing the adapters (as in this case) because we don't have any mechanism to work around this from an adapter. But, and more importantly, I also don't think that this is the correct behaviour semantically. I do think that "try to parse this value as x and ignore it otherwise" makes more sense for the decode functions. This more eager way of writing/understanding adapters also matches how we handle things in other places. Notably, we already gracefully handle empty values without the adapter having to concern itself with them:

https://github.com/tweaselORG/TrackHAR/blob/9287bf182bf18badb8ac93f0e300eec0b4f011c1/src/index.ts#L249

**Comment by baltpeter** [comment]
Implemented. The commit history should now be kept when merging.

**Review by zner0L — APPROVED** [review]

**Comment by baltpeter** [comment]
The last force push broke the ID grouping changes: [`183a5a0cd9b33ac18a465b6046573e4c06ed959b..71769496afc63164b7398852f7dc746b083abefd`#diff-6e40c2e22b](https://github.com/tweaselORG/TrackHAR/compare/183a5a0cd9b33ac18a465b6046573e4c06ed959b..71769496afc63164b7398852f7dc746b083abefd#diff-6e40c2e22bd2a3e07e99bc31f85d88156813c70d1558a35bf5cd4bab8040658a)L1632

I'm currently working on various Google adapters for web anyway, so I'll fix this through that PR.

**Pull request merged**